beamer/state_machine.py

```diff
@@ -278,13 +279,16 @@ def _handle_initiate_l1_resolution(
 
     assert request.fill_tx is not None, "Request not yet filled"
     if _l1_resolution_criteria_fulfilled(claim, context):
-        request.l1_resolution_tx_handle = context.resolution_pool.submit(
+        future = context.resolution_pool.submit(
             run_relayer,
             context.config.l1_rpc_url,
             context.config.l2b_rpc_url,
             context.config.account.key,
             request.fill_tx,
         )
+        context.l1_resolutions[request.id] = future
+        request.l1_resolution_started = True
```

Comment:
It seems we could use the `request.id in context.l1_resolutions` to find out whether L1 resolution has been started, thereby removing the need for `l1_resolution_started`.
However, having said that, I still think we should re-evaluate the request state machine and make this explicit in the state, because that is what our state machine serves for.

Reply:
> It seems we could use the `request.id in context.l1_resolutions` to find out whether L1 resolution has been started, thereby removing the need for `l1_resolution_started`.

That doesn't work, because we remove the future once it's finished.

> However, having said that, I still think we should re-evaluate the request state machine and make this explicit in the state, because that is what our state machine serves for.

This would duplicate most states, so I don't really like it.

Reply:
> That doesn't work, because we remove the future once it's finished.

Ah, right.
 
> > However, having said that, I still think we should re-evaluate the request state machine and make this explicit in the state, because that is what our state machine serves for.
> 
> This would duplicate most states, so I don't really like it.

We just had a call about this.
The idea here is not to duplicate any of the states, but just add a new one, e.g. `L1ResolutionInitiated`.
The new state could be entered only from `Claimed` and exited only to `Withdrawn`.
We would enter this new state upon seeing the relevant blockchain event, regardless of who actually initiated
L1 resolution.
Once the request is resolved, we simply do a withdraw, as usual.

Reply:
Done